### DISCUSSION TITLE
adaptive infill and temperature settings

### DISCUSSION THREAD

**Discussion by dxc-vv** [supermerill/SuperSlicer#2558] [Q&A]
I love superslicers, but some features are missing that we can find in other slicers and that seem very useful to me to introduce in superslicers:

adaptive infill (like ideaMaker)
possibility to set different temperature in base of a certain print level (like Simplify3D)

Do you think you can implement them? or are they already on the roadmap?

Many thanks.

**Comment by supermerill** [comment]
> adaptive infill (like ideaMaker)

prusa implemented some infiils pattern like [Adaptive cubic infill](https://help.prusa3d.com/en/article/infill-patterns_177130)

if what you want is a better support for the top solid layers, i advise to activated the 'dense infill' layer option in the infill tab. It will create a denser sparse support for solid layers, without any turn in mid-air -> no curling.

> possibility to set different temperature in base of a certain print level (like Simplify3D)

* add a height range modifier (click on the page/cog icon next to the object name in the right panel, or by the contextual menu when clicking on the object in the 3D view)
* set the start/end z for the range modifier
* right-click on the modifier's cog icon and select '`add settings` -> `filament` -> `temperature` -> `ok`
* then set your needed temperature for this region.
